### PR TITLE
Fix CSI plugin documentation link

### PR DESCRIPTION
**Pull request by JensErat**
Thank you for contributing to Velero!

# Please add a summary of your change

It feels like the CS plugin documentation link was copied from somewhere else, but not adjusted to the proper target.

# Does your change fix a particular issue?

I did not open an issue for the documentation bug.

# Please indicate you've done the following:

/kind changelog-not-required

- [X] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [X] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required`.
- [X] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by JensErat** [comment]
Has `/kind changelog-not-required` need to be added by a maintainer? I assume a minor documentation fixup is not changelog relevant.

File: site/content/docs/v1.6.0-rc.1/supported-providers.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ Velero supports a variety of storage providers for different backup and snapshot
...
 | [VMware vSphere](https://github.com/vmware-tanzu/velero-plugin-for-vsphere)              | 🚫                  | vSphere Volumes              | [VMware vSphere](https://github.com/vmware-tanzu/velero-plugin-for-vsphere)                    | [vSphere Plugin Setup](https://github.com/vmware-tanzu/velero-plugin-for-vsphere#installing-the-plugin)    |
-| [Container Storage Interface (CSI)](https://github.com/vmware-tanzu/velero-plugin-for-csi/)| 🚫                 | CSI Volumes                  | [Velero plugin for CSI](https://github.com/vmware-tanzu/velero-plugin-for-csi/)             | [CSI Plugin Setup](website-guidelines.md)        |
+| [Container Storage Interface (CSI)](https://github.com/vmware-tanzu/velero-plugin-for-csi/)| 🚫                 | CSI Volumes                  | [Velero plugin for CSI](https://github.com/vmware-tanzu/velero-plugin-for-csi/)             | [CSI Plugin Setup](https://github.com/vmware-tanzu/velero-plugin-for-csi#setup)        |
```
Comment (ashish-amarnath) [inline site/content/docs/v1.6.0-rc.1/supported-providers.md:18]: We are in the middle of a release and in the process this file will be moved to `site/content/docs/v1.6./supported-providers.md`

It'll be great if you can move this file post `v1.6` release.

**Review by ashish-amarnath — CHANGES_REQUESTED** [review]
Thanks for your help fixing up our docs @JensErat 
We are in the middle of the `v1.6` release, once we've released v1.6 can you please update the PR to move the `supported-providers.md` file in the directory corresponding to the v1.6 release.

File: site/content/docs/v1.6.0-rc.1/supported-providers.md (hunk 1 above)
Comment (jenting) [reply site/content/docs/v1.6.0-rc.1/supported-providers.md:18]: @JensErat could you please update it?

**Comment by ashish-amarnath** [comment]
@JensErat I am going to close out this PR. Please re-open when you are able to address the outstanding comment.

**Pull request closed without merging**